### PR TITLE
Improve the error message for getting all indexes of collection

### PR DESCRIPTION
**Pull request by yah01**
/kind improvement

### CODE REVIEW

File: internal/datacoord/index_service.go
Comment (congqixia) [inline internal/datacoord/index_service.go:1]: The actual API used when loading collection is `DescribeIndex`
`GetIndexState` is deprecated now

**Review by congqixia — CHANGES_REQUESTED** [review]
Don't think this could fix the problem, please check comment

File: pkg/util/merr/utils.go [hunk 1]
```diff
@@ -472,6 +472,14 @@ func WrapErrIndexNotFoundForSegment(segmentID int64, msg ...string) error {
 	return err
 }
 
+func WrapErrIndexNotFoundForCollection(collectionID int64, msg ...string) error {
```
Comment (congqixia) [inline pkg/util/merr/utils.go:475]: This error message contains collection id, which means nothing to user

**Review by congqixia — APPROVED** [review]
/lgtm

**Comment by yah01** [comment]
related #27490

File: internal/proxy/default_limit_reducer.go [hunk 2]
```diff
@@ -39,42 +38,38 @@ func (r *defaultLimitReducer) afterReduce(result *milvuspb.QueryResults) error {
...
+		field := typeutil.GetField(schema, outputFieldsID[i])
+		if field == nil {
+			err = merr.WrapErrFieldNotFound(outputFieldsID[i])
+			break
+		}
+
+		if result.FieldsData[i] == nil {
+			result.FieldsData[i], err = typeutil.GenEmptyFieldData(field)
+			if err != nil {
+				return err
```
Comment (congqixia) [inline internal/proxy/default_limit_reducer.go:60]: this branch return error directly without setting status like `field is nil` branch

File: internal/proxy/task.go [hunk 3]
```diff
@@ -8,7 +8,7 @@
 //
 //     http://www.apache.org/licenses/LICENSE-2.0
 //
-// Unless required by applicable law or agreed to in writing, software
+// Unless required by applicable law or agreed to in writing, sotware
```
Comment (congqixia) [inline internal/proxy/task.go:11]: why add a typo here?

File: internal/proxy/task_index.go [hunk 4]
```diff
@@ -491,13 +491,19 @@ func (dit *describeIndexTask) Execute(ctx context.Context) error {
...
+
 	dit.result = &milvuspb.DescribeIndexResponse{}
 	dit.result.Status = resp.GetStatus()
-	if dit.result.GetStatus().GetErrorCode() != commonpb.ErrorCode_Success {
-		return merr.Error(dit.result.GetStatus())
+	err = merr.Error(resp.GetStatus())
+	if err != nil {
+		if errors.Is(err, merr.ErrIndexNotFound) && len(dit.GetIndexName()) == 0 {
+			err = merr.WrapErrIndexNotFoundForCollection(dit.GetCollectionName())
+			dit.result.Status = merr.Status(err)
```
Comment (congqixia) [inline internal/proxy/task_index.go:504]: composing an error and convert it into status looks awkward here, maybe add a helper method later?

File: pkg/util/typeutil/schema.go [hunk 5]
```diff
@@ -833,6 +833,16 @@ func GetPrimaryFieldData(datas []*schemapb.FieldData, primaryFieldSchema *schema
...
 }
 
+func GetField(schema *schemapb.CollectionSchema, fieldID int64) *schemapb.FieldSchema {
+	for _, field := range schema.GetFields() {
+		if field.GetFieldID() == fieldID {
+			return field
+		}
+	}
+
+	return nil
```
Comment (congqixia) [inline pkg/util/typeutil/schema.go:843]: NIT: equivalent to `lo.Select(schema.GetFields(), func(...)`

File: internal/proxy/task_index.go (hunk 4 above)
Comment (yah01) [reply internal/proxy/task_index.go:504]: It's rewriting the error with the helper info

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**